instrumentation/kafka/kafka-clients/kafka-clients-2.6/library/src/test/java/io/opentelemetry/instrumentation/kafkaclients/v2_6/AbstractInterceptorsTest.java

```diff
@@ -28,19 +26,19 @@ abstract class AbstractInterceptorsTest extends KafkaClientBaseTest {
 
   static final String greeting = "Hello Kafka!";
 
+  protected abstract KafkaTelemetry kafkaTelemetry();
+
   @Override
   public Map<String, Object> producerProps() {
     Map<String, Object> props = super.producerProps();
-    props.put(
-        ProducerConfig.INTERCEPTOR_CLASSES_CONFIG, TracingProducerInterceptor.class.getName());
+    props.putAll(kafkaTelemetry().producerInterceptorConfigProperties());
```

Comment:
I guess the new way could fail when kafka properties are configured in a propeties file (idk, whether this is a thing at all) or when there is api built on top of kafka that lets you only set the class name.

Reply:
it's modeled after `OpenTelemetryMetricsReporter` which I think would also have this same problem